Rubberduck.Core/UI/UnitTesting/TestExplorerModel.cs

```diff
@@ -125,13 +86,198 @@ public bool IsBusy
+            {
+                Logger.Warn($"{(e.Test is null ? "Null" : "Unexpected")} test result handled by TestExplorerModel.");
+                return;
+            }
+
+            running.RunState = TestRunState.Running;
+        }
+
+        private void HandleRunCompletion(object sender, TestRunCompletedEventArgs e)
+        {
+            TotalDuration = e.Duration;
+
+            foreach (var test in Tests)
+            {
+                test.RunState = TestRunState.Stopped;
+            }
+
+            ExecutedCount = Tests.Count(t => t.Result.Outcome != TestOutcome.Unknown);
+            IsBusy = false;
+
+            OnPropertyChanged(nameof(LastTestRunSummary));
+            OnPropertyChanged(nameof(LastTestFailedCount));
+            OnPropertyChanged(nameof(LastTestIgnoredCount));
+            OnPropertyChanged(nameof(LastTestInconclusiveCount));
+            OnPropertyChanged(nameof(LastTestSucceededCount));
+        }
+
+        private void HandleTestCompletion(object sender, TestCompletedEventArgs e)
+        {
+            var test = e.Test;
+            var vmTest = new TestMethodViewModel(test);
+
+            if (!Tests.Contains(vmTest))
+            {
+                Tests.Add(vmTest);
+            }
+            else
+            {
+                vmTest = Tests.First(inside => inside.Equals(vmTest));
+            }
+
+            vmTest.RunState = TestRunState.Stopped;
+            vmTest.Result = e.Result;
+            
+            UpdateProgressBar(vmTest.Result.Outcome);
+            // Propagate the event.
+            OnTestCompleted(e);
+        }
+
+        private void HandleTestRefreshStarted(object sender, EventArgs args)
+        {
+            IsRefreshing = true;
+        }
+
+        private void HandleTestsRefreshed(object sender, EventArgs args)
+        {
+            IsRefreshing = false;
+
+            var previous = Tests.ToList();
+            
+            Tests.Clear();
+            foreach (var test in _testEngine.Tests)
+            {
+                var adding = new TestMethodViewModel(test);
+                var match = previous.FirstOrDefault(ut => ut.Method.Equals(test));
+                if (match != null)
+                {
+                    adding.Result = match.Result;
+                }
+                Tests.Add(adding);
+            }
+        }
+
+        private void OnTestCompleted(TestCompletedEventArgs args)
+        {
+            TestCompleted?.Invoke(this, args);
+        }
+
+        private static readonly Dictionary<TestOutcome, Color> OutcomeColors = new Dictionary<TestOutcome, Color>
+        {
+            { TestOutcome.Unknown, Colors.DimGray },
+            { TestOutcome.Succeeded, Colors.LimeGreen },
+            { TestOutcome.Inconclusive, Colors.Gold },
+            { TestOutcome.Ignored, Colors.Orange },
+            { TestOutcome.Failed, Colors.Red }
+        };
+
+        private TestOutcome _bestOutcome = TestOutcome.Unknown;
```

Comment:
"Best" implies that if a single test passes with 99 failures, we get green. I'm pretty sure that's not the "best" outcome. Maybe `_worstOutcome`? `_lcdOutcome`?

Reply:
Doh! I somehow missed that when the logic flipped.  Will fix.